diff -up qtmultimedia-everywhere-src-5.15.2/src/backends/avplayer/qavplayer.cpp.omv~ qtmultimedia-everywhere-src-5.15.2/src/backends/avplayer/qavplayer.cpp
--- qtmultimedia-everywhere-src-5.15.2/src/backends/avplayer/qavplayer.cpp.omv~	2021-04-03 17:54:16.170904625 +0200
+++ qtmultimedia-everywhere-src-5.15.2/src/backends/avplayer/qavplayer.cpp	2021-04-03 18:06:53.210806764 +0200
@@ -264,9 +264,9 @@ void QAVPlayerPrivate::doLoad(const QUrl
         setMediaStatus(QMediaPlayer::LoadedMedia);
     });
 
-    demuxerFuture = QtConcurrent::run(&QAVPlayerPrivate::doDemux, this);
-    videoPlayFuture = QtConcurrent::run(&QAVPlayerPrivate::doPlayVideo, this);
-    audioPlayFuture = QtConcurrent::run(&QAVPlayerPrivate::doPlayAudio, this);
+    demuxerFuture = QtConcurrent::run(this, &QAVPlayerPrivate::doDemux);
+    videoPlayFuture = QtConcurrent::run(this, &QAVPlayerPrivate::doPlayVideo);
+    audioPlayFuture = QtConcurrent::run(this, &QAVPlayerPrivate::doPlayAudio);
 }
 
 void QAVPlayerPrivate::doDemux()
@@ -439,7 +439,7 @@ void QAVPlayer::setSource(const QUrl &ur
 
     d->quit = false;
     d->setMediaStatus(QMediaPlayer::LoadingMedia);
-    d->loaderFuture = QtConcurrent::run(&QAVPlayerPrivate::doLoad, d, d->url);
+    d->loaderFuture = QtConcurrent::run(d, &QAVPlayerPrivate::doLoad, d->url);
     QMutexLocker locker(&d->waitMutex);
     d->wait = true;
 }
diff -up qtmultimedia-everywhere-src-5.15.2/src/backends/backends.pro.omv~ qtmultimedia-everywhere-src-5.15.2/src/backends/backends.pro
--- qtmultimedia-everywhere-src-5.15.2/src/backends/backends.pro.omv~	2021-04-03 19:12:04.842597788 +0200
+++ qtmultimedia-everywhere-src-5.15.2/src/backends/backends.pro	2021-04-03 19:12:10.335904145 +0200
@@ -1,6 +1,6 @@
 TEMPLATE = subdirs
 
-include($$OUT_PWD/multimedia/qtmultimedia-config.pri)
+include($$OUT_PWD/../multimedia/qtmultimedia-config.pri)
 QT_FOR_CONFIG += multimedia-private
 
 qtConfig(ffmpeg): SUBDIRS += avplayer
